src/main/java/com/amazonaws/sagemaker/controller/ServingController.java

```diff
@@ -148,6 +150,54 @@ public ResponseEntity returnBatchExecutionParameter() throws JsonProcessingExcep
         }
     }
 
+    /**
+     * Implements the invocations POST API for application/jsonlines input
+     *
+     * @param jsonLines, lines of json values
+     * @param accept, accept parameter from request
+     * @return ResponseEntity with body as the expected payload JSON & proper statuscode based on the input
+     */
+    @RequestMapping(path = "/invocations", method = POST, consumes = AdditionalMediaType.APPLICATION_JSONLINES_VALUE)
+    public ResponseEntity<String> transformRequestJsonLines(@RequestBody final byte[] jsonLines,
+                                                            @RequestHeader(value = HttpHeaders.ACCEPT, required = false) final String accept) {
+        if (jsonLines == null) {
+            LOG.error("Input passed to the request is empty");
+            return ResponseEntity.noContent().build();
+        }
+        try {
+            final String acceptVal = this.retrieveAndVerifyAccept(accept);
+            final DataSchema schema = this.retrieveAndVerifySchema(null, mapper);
+            final String jsonStringLine = new String(jsonLines);
+
+            // Map list of inputs to DataList object
+            final SageMakerDataListObject sro = mapper.readValue(jsonStringLine, SageMakerDataListObject.class);
+            List<List<Object>> inputDatas = sro.getData();
+            List<ResponseEntity<String>> responseList = Lists.newArrayList();
+
+            // Process each input separately and add response to a list
+            final int inputDatasSize = inputDatas.size();
+            for (int idx = 0; idx < inputDatasSize; ++idx) {
+                ResponseEntity<String> response = this.processInputData(inputDatas.get(idx), schema, acceptVal);
+                responseList.add(response);
+            }
+
+            // Merge response body to a new output response
+            List<List<String>> bodyList = Lists.newArrayList();
+            HttpHeaders headers = null;
+            //combine body in responseList
+            for (ResponseEntity<String> response:responseList) {
+                HttpStatus statuscode = response.getStatusCode();
+                headers = response.getHeaders();
+                bodyList.add(Lists.newArrayList(response.getBody()));
+            }
+
+            return ResponseEntity.ok().headers(headers).body(bodyList.toString());
+        } catch (final Exception ex) {
+            LOG.error("Error in processing current request", ex);
+            return ResponseEntity.badRequest().body(ex.getMessage());
```

Comment:
Will ex.getMessage() output the stack trace as well? How about you just pass in ex to the body? You might want to test that.

Reply:
ex is Exception type but here responseEntity requires the body as String so we may use getMessage here.